lib/segment/src/common/reciprocal_rank_fusion.rs

```diff
@@ -0,0 +1,168 @@
+use std::collections::hash_map::Entry;
+
+use ahash::{HashMap, HashMapExt};
+use ordered_float::OrderedFloat;
+
+use crate::types::{ExtendedPointId, ScoredPoint};
+
+/// Reciprocal Rank Fusion (RRF) is a method for combining rankings from multiple sources.
+/// See https://plg.uwaterloo.ca/~gvcormac/cormacksigir09-rrf.pdf
+
+/// Mitigates the impact of high rankings by outlier systems
+const RFF_RANKING_K: f32 = 2.0;
+
+/// Compute RRF scores for multiple results from different sources.
+/// Each response can have a different length.
+/// Each response must be sorted by score in descending order.
+///
+/// The output is a single sorted list of ScoredPoint.
+/// Does not break ties.
+#[allow(dead_code)]
+pub fn rrf_scoring(responses: Vec<Vec<ScoredPoint>>, limit: usize) -> Vec<ScoredPoint> {
+    // ensure each input is sorted
+    debug_assert!(responses
+        .iter()
+        .all(|r| r.windows(2).all(|w| w[0].score >= w[1].score)));
```

Comment:
This part might need to take into account Order::LargeBetter, or Order::SmallBetter

Reply:
I removed this assert as it does not bring much value.